src/pushsource/_impl/backend/koji_source.py

```diff
@@ -621,6 +622,97 @@ def _push_items_from_vmi_build(self, nvr, meta):
 
         return out
 
+    def _push_items_from_rhcos_build(self, nvr, meta, archives=None):
+
+        LOG.debug("Looking for core os assembler build on %s, %s", nvr, meta)
+
+        build_id = meta["id"]
+        archives = archives or self._get_archives(build_id)
+
+        for archive in archives:
+            if archive.get("filename") == "meta.json":
+                path = self._pathinfo.typedir(meta, archive["btype"])
+                item_src = os.path.join(path, archive["filename"])
+                with open(item_src) as archive_file:
+                    rhcos_meta_data = json.loads(archive_file.read())
+
+        extra = meta.get("extra") or {}
+        image = extra.get("typeinfo").get("image") or {}
+
+        arch = image.get("arch")
+        boot_mode = image.get("boot_mode")
+
+        version = rhcos_meta_data["coreos-assembler.container-config-git"][
```

Comment:
Is this structure documented anywhere?

This code doesn't tolerate a missing field at all, it'll just crash if the field isn't there.  If the owners of the tool generating this field haven't published a doc with the fields they're producing, I'm concerned that they'll feel free to change it at any time.

The concern would be mitigated if this could be written in such a way that it wouldn't crash.

Reply:
https://github.com/coreos/coreos-assembler/blob/efa00a6b07431bbfdf7a80032612290aea6a62bc/src/v1.json#L244
This file have the schema for rhcos meta.json file, but it seems that they have marked "coreos-assembler.container-config-git" as optional. I have given one 'or' block. let me know if it looks good or not.